src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs

```diff
@@ -0,0 +1,521 @@
+            if (InteractionAllowed)
+            {
+                AzureDevOps.Disconnect();
+
+                ClearTreeviewSelection();
+                ChangeStates(States.NoServer);
+                AzureDevOps.Configuration.SavedConnection = null;
+            }
+
+            canSave = false;
+            Dispatcher.Invoke(UpdateSaveButton);
+        }
+
+        /// <summary>
+        /// Refresh the project teams, equivalent to reloading this state
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void refreshButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Don't allow refresh until project teams have loaded
+            if (InteractionAllowed)
+            {
+                this.tbTeamProjectSearch.Clear();
+                ClearTreeviewSelection();
+                ChangeStates(States.EditingServer);
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Switch screens in this connection config view
+        /// </summary>
+        /// <param name="state"></param>
+        private async void ChangeStates(States state)
+        {
+            Dispatcher.Invoke(() => { 
+                vmAvatar.ByteData = AzureDevOps.Avatar;
+                displayNameField.Text = AzureDevOps.DisplayName;
+                emailField.Text = AzureDevOps.Email;
+            });
+            if (state == States.NoServer)
+                {
+                Dispatcher.Invoke(() =>
+                {
+                    selectServerGrid.Visibility = Visibility.Visible;
+                    editProjectGrid.Visibility = Visibility.Collapsed;
+                    teamSelectedGrid.Visibility = Visibility.Collapsed;
+                });
+                }
+                else if (state == States.EditingServer)
+                {
+                Dispatcher.Invoke(() =>
+                {
+                    editProjectGrid.Visibility = Visibility.Visible;
+                    selectTeamGrid.Visibility = Visibility.Visible;
+                    teamSelectedGrid.Visibility = Visibility.Collapsed;
+                    selectServerGrid.Visibility = Visibility.Collapsed;
+                });
+                    // Load the team project list, show progress animation
+                    try
+                    {
+                        ToggleLoading(true);
+                        projects.Clear();
+                        var newProjectList = await UpdateTeamProjects().ConfigureAwait(true); // need to come back to original UI thread. 
+                        newProjectList.ForEach(p => projects.Add(p));
+                        ToggleLoading(false);
+                    Dispatcher.Invoke(() => serverTreeview.ItemsSource = projects);
+                    }
+                    catch (Exception)
+                    {
+                        //MessageDialog.Show("Error when retrieving team projects");
```

Comment:
It does look like we want to show this particular error to the user. This can also be done withing the extension UI.

Reply:
Yes, I still need to port over MessageDialog